extensions/wikia/AbPerformanceTesting/classes/Hooks.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Wikia\AbPerformanceTesting;
+
+use \Wikia\Logger\WikiaLogger;
+
+class Hooks {
+
+	/**
+	 * Initialize the experiment and set all required tracking things
+	 *
+	 * @param string $experimentName
+	 * @param array $experimentConfig
+	 */
+	private static function startExperiment( $experimentName, Array $experimentConfig ) {
+		wfDebug( sprintf( "%s[%s] using %s class with %s params\n",
+			__METHOD__, $experimentName, $experimentConfig['handler'], json_encode( $experimentConfig['params'] ) ) );
+
+		new $experimentConfig['handler']( $experimentConfig['params'] ? : [] );
+
+		// mark a transaction with an experiment name
+		\Transaction::getInstance()->set( \Transaction::PARAM_AB_PERFORMANCE_TEST, $experimentName );
+
+		// mark a transaction using UA's custom dimensions
+		global $wgHooks;
+		$wgHooks['WikiaSkinTopScripts'][] = function( Array &$vars, &$scripts ) use ( $experimentName ) {
+			$val = \Xml::encodeJsVar( $experimentName );
+			$scripts .= \Html::inlineScript( "_gaq.push(['set', 'dimension20', {$val}]);" );
```

Comment:
Oh and one more thing. Above code will set dimension on the main property so it will be 10% sampled. If you want to be also collected for spacial wiki's (100% sampled) add this:
```
ga('special.set', 'dimension20', {$val});
```